ecosystem/indexer-grpc/indexer-grpc-table-info/src/internal_indexer_db_service.rs

```diff
@@ -59,8 +59,53 @@ impl InternalIndexerDBService {
         Arc::clone(&self.db_indexer)
     }
 
-    pub async fn run(&mut self) {
-        let mut start_version = self.db_indexer.get_persisted_version().unwrap_or(0);
+    pub fn get_start_version(&self, node_config: &NodeConfig) -> Version {
+        let indexer_version = self
+            .db_indexer
+            .get_persisted_version()
+            .unwrap_or_else(|err| {
+                panic!(
+                    "Failed to get persisted version from internal indexer db: {}",
+                    err
+                )
+            });
```

Comment:
I feel we should return error here (and many other places) and panic at the top level of the indexer task.